### PR TITLE
Add support for named pipes

### PR DESCRIPTION
**Pull request by Nino-K**
Signed-off-by: Nino Kodabande <nkodabande@suse.com>

### CODE REVIEW

File: src/go/vtunnel/test/e2e/connectivity_test.go [hunk 1]
```diff
@@ -174,19 +233,102 @@ func TestConnect(t *testing.T) {
...
+		err = downloadFile(tarballPath, wslTarballURL)
+		requireNoErrorf(err, "Failed to download wsl distro tarball %v", wslTarballName)
+	}
+	if s.ModTime().Before(time.Now().Add(time.Hour * -1)) {
+		err = downloadFile(tarballPath, wslTarballURL)
+		requireNoErrorf(err, "Failed to download wsl distro tarball after cache expiry %v", wslTarballName)
+	}
+
+	logrus.Infof("Creating %v wsl distro", wslDistroName)
+	installCmd := cmdExec(
```
Comment (mook-as) [inline src/go/vtunnel/test/e2e/connectivity_test.go:280]: Do we need `installCmd` (as opposed to `cmdExec(…).Run()` directly)?

File: src/go/vtunnel/test/e2e/connectivity_test.go [hunk 2]
```diff
@@ -174,19 +233,102 @@ func TestConnect(t *testing.T) {
...
+		wslDistroName,
+		".",
+		tarballPath)
+	err = installCmd.Run()
+	requireNoErrorf(err, "Failed to install distro %v", wslDistroName)
+
+	// It takes a long time to start a new distro,
+	// 20 sec is a long time but that's actually how long
+	// it takes to start a distro without any flakiness
+	timeout := time.Second * 20
```
Comment (mook-as) [inline src/go/vtunnel/test/e2e/connectivity_test.go:293]: Given that we're just polling in a loop, the timeout length is probably irrelevant — even going to a full minute is fine.

File: src/go/vtunnel/test/e2e/connectivity_test.go [hunk 3]
```diff
@@ -174,19 +233,102 @@ func TestConnect(t *testing.T) {
...
+	l, err := winio.ListenPipe(uri[len("npipe://"):], nil)
+	require.NoError(t, err, "Failed to listen on named pipe: %v", nPipeEndpoint)
+	for {
+		c, err := l.Accept()
+		if err != nil {
+			require.NoError(t, err, "accepting incoming named pipe connection: %w")
+		}
+		_, err = c.Write([]byte(fmt.Sprintf("vtunnel named pipe %v called.", uri)))
+		require.NoError(t, err, "Failed to write to named pipe: %v", nPipeEndpoint)
+		time.Sleep(time.Second * 2)
```
Comment (mook-as) [inline src/go/vtunnel/test/e2e/connectivity_test.go:327]: Please add a comment on this sleep.

**Review by mook-as — CHANGES_REQUESTED** [review]
I was unable to run the tests due to a crash; please it (identified) and we can try again.

**Review by mook-as — APPROVED** [review]

**Comment by Nino-K** [comment]
https://github.com/rancher-sandbox/rancher-desktop/issues/2677

**Pull request merged**